refresh.py

```diff
@@ -0,0 +1,180 @@
+"""
+This script parses a token.xml file, collects the picURLs within, and replaces
+the URLs to Scryfall cards with up-to-date URLs by querying Scryfall's API.
+"""
+
+from xml.sax import saxutils, make_parser, handler
+from urllib.parse import urlsplit
+from urllib.request import Request, urlopen
+import itertools
+import json
+import sys
+import time
+import os
+import tempfile
+import pathlib
+import shutil
+
+SCRYFALL_MAX_LIST_SIZE = 75
+
+def cards_collection(identifiers):
+    """
+    Get information about a set of cards using the Scryfall API.
+
+    This simply returns a list of dictionaries representing the Card objects as
+    returned by the /card/collection Scryfall API.
+
+    If the list of identifiers is larger than Scryfall's API limit,
+    cards_collection automatically splits the list into smaller chunks and
+    makes multiple requests.
+    """
+
+    start_time = 0
+    n = 0
+    while n < len(identifiers):
+        chunk = identifiers[n:n + SCRYFALL_MAX_LIST_SIZE]
+        print("Requesting chunk {}-{}/{}...".format(n, n + len(chunk), len(identifiers)))
+        n += SCRYFALL_MAX_LIST_SIZE
+
+        payload = json.dumps({'identifiers': chunk}).encode('utf-8')
+        req = Request('https://api.scryfall.com/cards/collection', payload,
+                      headers={'Content-Type': 'application/json'})
+
+        # Rate limiting
+        cur_time = time.time()
+        delta_time = cur_time - start_time
+        if delta_time < 0.1:
+            time.sleep(0.1 - delta_time)
+        start_time = time.time()
+
+        with urlopen(req) as f:
+            list_obj = json.load(f)
+            assert not list_obj.get('has_more', False)
+            assert 'warnings' not in list_obj
+
+            yield from list_obj['data']
+
+def parse_picurl(picurl):
+    """
+    Parse a Scryfall picURL into its components.
+
+    The Scryfall picURL must be in one of those forms:
+
+        - https://c1.scryfall.com/file/scryfall-cards/<version>/<face>/*/*/<uuid>.jpg
+        - https://cards.scryfall.io/<version>/<face>/*/*/<uuid>.jpg
```

Comment:
Do we need to account for any of the other variants?
https://scryfall.com/blog/upcoming-api-changes-to-scryfall-image-uris-and-download-uris-224

Reply:
Not sure. It looks like only cards.scryfall.io is used for cards currently.

Maybe it would make sense for the script to check that it can re-parse the resolved URL and print out an error otherwise? This would indicate a change in scryfall's URLs that needs to be reflected in the script.